brianlong/www.validators.app
# 185232287 - fix api validator response

#### What's this PR do?
Story fixes api validator(index) response, in terms of valid validator_histories(its epoch especially).
It applies additional fix to query association between validator - validator_histories.

#### How should this be manually tested?
Make sure you have various ValidatorHistory records with the same account value for different networks. Something like:
```
create(:validator_history, account: @account, network: "testnet")
create(:validator_history, account: @account, network: "testnet")
create(:validator_history, account: @account, network: "mainnet")
```
- verify if the validator returns proper records for its association `has_many :validator_histories`
- check API response for index action (e.g. `api/v1/validators/mainnet.json?q=9cS2cDhLmRoUCqkhMjXZs4HC1YYbP5QYweCP3vVqodcb`). This endpoint should return objects related to specific network only
- investigate `https://stage.validators.app/api/v1/validators/:network.json` for any performance issues.

#### What are the relevant tickets?
[- URL to the PivotalTracker ticket](https://www.pivotaltracker.com/story/show/185232287)

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [N] Does this PR have any migrations?
- [N] Does this PR add new dependencies?
